x-pack/plugins/snapshot_restore/public/app/sections/home/home.tsx

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { PureComponent } from 'react';
+import { Route, RouteComponentProps, Switch } from 'react-router-dom';
+
+import { EuiPageBody, EuiPageContent, EuiSpacer, EuiTab, EuiTabs, EuiTitle } from '@elastic/eui';
+
+import { BASE_PATH } from '../../constants';
+import { AppContext, AppContextInterface } from '../../services/app_context';
+
+import { RepositoryList } from './repository_list';
+import { SnapshotList } from './snapshot_list';
+
+type Section = 'repositories' | 'snapshots';
+
+interface MatchParams {
+  section: Section;
+}
+
+interface Props extends RouteComponentProps<MatchParams> {}
+
+interface State {
+  activeSection: Section;
+}
+
+export class SnapshotRestoreHome extends PureComponent<Props, State> {
+  public static contextType = AppContext;
+
+  public static getDerivedStateFromProps(nextProps: Props) {
```

Comment:
wondering if maybe we want to start using function components + hooks instead of class ones.

Reply:
I'd like to address this with the next pass when actual features start getting built out, it's hard for me to formulate an opinion for function components with hooks instead of lifecycles when the components don't do much yet 😁 I'll try function components + hooks for new components, and refactor these class ones if it goes well!